File: CppPractice/CppLeetcode/Leetcode148.cpp
class Solution {
public:
    ListNode* sortList(ListNode* head) 
    {
        ListNode *cur = head;
        vector<int> numList;
        while(cur)
        {
            numList.push_back(cur->val);
            cur=cur->next;
        }
        sort(numList.begin(),numList.end());
        cur =head;
        int curIndex = 0;
        while(cur)
        {
            cur->val = numList[curIndex];
            cur = cur->next;
            curIndex++;
        }
        return head;
    }
};

File: CSharpPractice/LeetcodePractice/Leetcode148.cs
public class Solution148 {
    public ListNode SortList(ListNode head) {
        List<int> numList = new List<int>();
        ListNode cur = head;
        while(cur!=null)
        {
            numList.Add(cur.val);
            cur = cur.next;
        }
        numList.Sort();
        cur = head;
        int curIndex=0;
        while(cur!=null)
        {
            cur.val  = numList[curIndex];
            curIndex++;
            cur=cur.next;
        }
        return head;
    }
}